### ISSUE TITLE
Menu: menu does not close, when clicking outside

### ISSUE THREAD

**Issue by smoshko1612** [primefaces/primeng#13381]
### Describe the bug

A put the menu with button for expand it in the table cell component. When I click on button menu is open. But when I click on another button the menu is not closed.

### Environment

Chrome

### Angular version

15.1.0

### PrimeNG version

11.4.0

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

16.16.0

### Browser(s)

Chrome

### Steps to reproduce the behavior

1. Create table component
2. Add menu and expand button inside
3. click on the 3 menu buttons.
actual: 3 menu are visible.

### Expected behavior

expected: Only one menu should be shown.

**Comment by smoshko1612** [comment]
![image](https://github.com/primefaces/primeng/assets/136705946/6adf5b8d-57ff-424e-8bfa-99bb937e7a1c)

**Comment by cetincakiroglu** [comment]
Hi, 

Cannot replicate the issue could you please share a Stackblitz example?

Here is a working example: https://stackblitz.com/edit/ipfylh?file=src%2Fapp%2Fdemo%2Ftable-basic-demo.html,src%2Fapp%2Fdemo%2Ftable-basic-demo.ts

**Comment by cetincakiroglu** [comment]
Closing the issue since it belongs to an outdated version and cannot be reproducible in the latest, please use the newer versions of the PrimeNG.

**Comment by TheVikingTaco** [comment]
Can we re-open this one, I'm seeing the same problem as this person in version 17

![Image](https://github.com/user-attachments/assets/ec73186d-1d6f-46ec-a5b3-c7c1b7567c55)

Same problem as the person the created the bug but in my case, I created a card component instead of a table, this component has another component that has the p-menu component.

as with the original post, everytime that I click on the three-dot button the menu shows up and can have as many menus open as I have cards.

**Comment by Celebes** [comment]
I had the same problem in PrimeNG 19. My template looked something like this:

```
@if (menuItems(); as menuItems) {
  <p-menu #menu [model]="menuItems" [popup]="true" appendTo="body"/>
  <p-button label="Actions"
            icon="fa-solid fa-chevron-down"
            variant="outlined"
            iconPos="right"
            severity="contrast"
            [disabled]="disabled()"
            (onClick)="$event.stopPropagation();menu.toggle($event);"/>
}
```

I've added `$event.stopPropagation();` because the actions button was used inside a p-table row, but as it turns out - it was unnecessary and the cause of a bug.

tl;dr - make sure you don't consume mouse event via `$event.stopPropagation();` in your menu trigger
